# Description: Add additional checks for features before running
#  We used to ship a shell script that checked for various things before
#  running compiz. However, this duplicated several checks that compiz
#  runs on its own and was slow due to being a shell script.
#  .
#  This patch adds in default options and plugins if none are provided on
#  the command line so running `compiz` continues to work like expected.
# Forwarded: not-needed
# Origin: Ubuntu
# Author: Travis Watkins <amaranth@ubuntu.com>

Index: compiz-0.9.4/src/main.cpp
===================================================================
--- compiz-0.9.4.orig/src/main.cpp	2011-02-24 10:27:13.000000000 +0100
+++ compiz-0.9.4/src/main.cpp	2011-02-24 11:17:38.142752891 +0100
@@ -170,6 +170,10 @@
 	}
     }
 
+    /* add in default plugins if none are given */
+    if (plugins.size () == 0)
+	plugins.push_back ("ccp");
+
     initialPlugins = plugins;
 
     return true;
@@ -226,7 +230,16 @@
 	return false;
 
     if (!disableSm)
+    {
+	if (clientId == NULL)
+	{
+	    char *desktop_autostart_id = getenv ("DESKTOP_AUTOSTART_ID");
+	    if (desktop_autostart_id != NULL)
+		clientId = strdup (desktop_autostart_id);
+	    unsetenv ("DESKTOP_AUTOSTART_ID");
+	}
 	CompSession::init (clientId);
+    }
 
     return true;
 }
